llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4142,6 +4119,29 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpVariable>(SPIRVValue *con
       readOnly = true;
   }
 
+  if (!readOnly && storageClass == StorageClassFunction) {
```

Comment:
In case it's not obvious: the only change here is moving this block down to below the calculation of `readOnly`, and then adding `!readOnly &&` to the condition.